src/modules/esl-avatar/core/esl-avatar.ts

```diff
@@ -0,0 +1,109 @@
+import {ExportNs} from '../../esl-utils/environment/export-ns';
+import {ESLBaseElement} from '../../esl-base-element/core';
+import {attr, bind, boolAttr, prop} from '../../esl-utils/decorators';
+
+@ExportNs('Avatar')
+export class ESLAvatar extends ESLBaseElement {
+  public static override is = 'esl-avatar';
+  public static observedAttributes = ['image-src', 'username'];
+
+  /** Event to dispatch on change of {@link ESLAvatar} */
+  @prop('esl:avatar:changed') public AVATAR_CHANGED_EVENT: string;
+
+  /** Source path of the avatar image */
+  @attr() public imageSrc: string;
+  /** The limit number of letters to be displayed in text-only mode */
+  @attr({defaultValue: 2}) public limit: number;
```

Comment:
To Discuss (minor): the name of the attribute is not clear enough
Possibly: `max-char`, `truncate`, `abbr-length`, just `length`